src/meta/processors/usersMan/AuthenticationProcessor.cpp

```diff
@@ -0,0 +1,271 @@
+        if (!checkPassword(userRet.value(), req.get_old_encoded_pwd())) {
+            resp_.set_code(cpp2::ErrorCode::E_INVALID_PASSWORD);
+            onFinished();
+            return;
+        }
+    }
+
+    auto userKey = MetaServiceUtils::userKey(userRet.value());
+    std::string val;
+    auto result = kvstore_->get(kDefaultSpaceId_, kDefaultPartId_, userKey, &val);
+    if (result != kvstore::ResultCode::SUCCEEDED) {
+        resp_.set_code(cpp2::ErrorCode::E_NOT_FOUND);
+        onFinished();
+        return;
+    }
+    std::vector<kvstore::KV> data;
+    data.emplace_back(std::move(userKey),
+                      MetaServiceUtils::changePassword(val, req.get_new_encoded_pwd()));
+    resp_.set_code(cpp2::ErrorCode::SUCCEEDED);
+    doPut(std::move(data));
+}
+
+
+void GetUserProcessor::process(const cpp2::GetUserReq& req) {
+    folly::SharedMutex::ReadHolder rHolder(LockUtils::userLock());
+    auto userRet = getUserId(req.get_account());
+    if (!userRet.ok()) {
+        LOG(ERROR) << "User " << req.get_account() << " not found.";
+        resp_.set_code(cpp2::ErrorCode::E_NOT_FOUND);
+        onFinished();
+        return;
+    }
+    auto userKey = MetaServiceUtils::userKey(userRet.value());
+    std::string val;
+    auto result = kvstore_->get(kDefaultSpaceId_, kDefaultPartId_, userKey, &val);
+    if (result != kvstore::ResultCode::SUCCEEDED) {
+        resp_.set_code(cpp2::ErrorCode::E_NOT_FOUND);
+        onFinished();
+        return;
+    }
+    decltype(resp_.user_item) user = MetaServiceUtils::parseUserItem(val);
+    resp_.set_user_item(user);
+    resp_.set_code(cpp2::ErrorCode::SUCCEEDED);
+    onFinished();
+}
+
+
+void ListUsersProcessor::process(const cpp2::ListUsersReq& req) {
+    UNUSED(req);
+    folly::SharedMutex::ReadHolder rHolder(LockUtils::userLock());
+    std::unique_ptr<kvstore::KVIterator> iter;
+    std::string prefix = "__users__";
+    auto ret = kvstore_->prefix(kDefaultSpaceId_, kDefaultPartId_, prefix, &iter);
+    if (ret != kvstore::ResultCode::SUCCEEDED) {
+        LOG(ERROR) << "Can't find any users.";
+        resp_.set_code(cpp2::ErrorCode::E_NOT_FOUND);
+        onFinished();
+        return;
+    }
+    decltype(resp_.users) users;
+    while (iter->valid()) {
+        cpp2::UserItem user = MetaServiceUtils::parseUserItem(iter->val());
+        auto userId = MetaServiceUtils::parseUserId(iter->key());
+        users.emplace(userId, std::move(user));
+        iter->next();
+    }
+    resp_.set_users(users);
+    resp_.set_code(cpp2::ErrorCode::SUCCEEDED);
+    onFinished();
+}
+
+
+void CheckPasswordProcessor::process(const cpp2::CheckPasswordReq& req) {
+    folly::SharedMutex::ReadHolder rHolder(LockUtils::userLock());
+    auto userRet = getUserId(req.get_account());
+    if (!userRet.ok()) {
+        resp_.set_code(to(userRet.status()));
+        onFinished();
+        return;
+    }
+
+    if (!checkPassword(userRet.value(), req.get_encoded_pwd())) {
```

Comment:
I want to know when to check the password?

Reply:
This is a reserved interface，Used for client or terminal connections.